### PR TITLE
introduce package samtools v 1.2

### PR DESCRIPTION
**Pull request by martenson**
I copied the 1.1, took @davebx binary and added action to remove the /usr/local prefix to the recipe (also by Dave) :)

### CODE REVIEW

File: packages/package_samtools_1_2/tool_dependencies.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                        <repository name="package_ncurses_5_9" owner="iuc">
+                            <package name="ncurses" version="5.9" />
+                        </repository>
+                        <repository name="package_zlib_1_2_8" owner="iuc">
+                            <package name="zlib" version="1.2.8" />
+                        </repository>
+                    </action>
+                    <action type="shell_command">sed -i.bak 's#/usr/local#$INSTALL_DIR#' Makefile</action>
+                    <action type="shell_command">sed -i -e 's/-lcurses/-lncurses/' Makefile</action>
+                    <action type="shell_command">sed -i -e "s|CFLAGS=\s*-g\s*-Wall\s*-O2\s*|CFLAGS= -g -Wall -O2 -I$NCURSES_INCLUDE_PATH/ncurses/ -I$NCURSES_INCLUDE_PATH -L$NCURSES_LIB_PATH|g" Makefile</action>
```
Comment (bgruening) [inline packages/package_samtools_1_2/tool_dependencies.xml]: @martenson do you know if this hack is still needed with the new samtools >= 1.0?

Comment (martenson) [inline packages/package_samtools_1_2/tool_dependencies.xml]: I do not.
@davebx Do you?

Comment (davebx) [inline packages/package_samtools_1_2/tool_dependencies.xml]: I'm not sure about the CFLAGS, samtools compiled fine locally without it. The substitution of ncurses for curses is still not a bad idea, I think.

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/43/
Test (probably) passed

**Comment by peterjc** [comment]
Can someone now push ``package_samtools_1_2`` to the Test Tool Shed, and if that looks good, the Main Tool Shed? Thanks!

(I have a couple of wrappers which can try using this, once the Tool Shed testing service is working again)

**Comment by martenson** [comment]
I am doing that right now.

**Comment by peterjc** [comment]
Thanks - real time GitHub notifications via email have their downsides ;)

**Pull request merged**